bundles/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchHandler.java

```diff
@@ -0,0 +1,236 @@
+                if (localConnector == null) {
+                    updateStatus(ThingStatus.OFFLINE);
+                    return;
+                }
+                localConnector.getValue("A1");
+            } catch (IOException io) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, io.toString());
+                return;
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.toString());
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+        });
+
+        // start refresh handler
+        startAutomaticRefresh();
+    }
+
+    private void startAutomaticRefresh() {
+        var localRefreshJob = refreshJob;
+        if (localRefreshJob == null || localRefreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+                    Set<String> tags = new HashSet<String>();
+                    for (EcoTouchTags ecoTouchTag : EcoTouchTags.values()) {
+                        String channel = ecoTouchTag.getCommand();
+                        boolean linked = isLinked(channel);
+                        if (linked)
+                            tags.add(ecoTouchTag.getTagName());
+                    }
+                    var localConnector = connector;
+                    if (localConnector != null) {
+                        Map<String, String> result = localConnector.getValues(tags);
+
+                        Iterator<Map.Entry<String, String>> it = result.entrySet().iterator();
+                        while (it.hasNext()) {
+                            Map.Entry<String, String> pair = it.next();
+                            updateChannel(pair.getKey(), pair.getValue());
+                        }
+                    }
+                } catch (IOException io) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, io.toString());
+                } catch (Exception e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.toString());
+                } catch (Error e) {
+                    // during thing creation, the following error is thrown:
+                    // java.lang.NoSuchMethodError: 'org.openhab.binding.ecotouch.internal.EcoTouchTags[]
+                    // org.openhab.binding.ecotouch.internal.EcoTouchTags.values()'
+                    // not sure why... lets ignore it for now
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            };
+
+            var localConfig = config;
+            if (localConfig != null)
+                refreshJob = scheduler.scheduleWithFixedDelay(runnable, 10, localConfig.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        var localRefreshJob = refreshJob;
+        if (localRefreshJob != null && !localRefreshJob.isCancelled()) {
```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (localRefreshJob != null) {
```